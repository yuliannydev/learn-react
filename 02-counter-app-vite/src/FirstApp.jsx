import React from 'react';
import PropTypes from 'prop-types';
export default function FirstApp({title, subTitle, name}) {
  return (
    <>
    <h1> {title} </h1>
    <p> {subTitle} </p>
    <em> {name} </em>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
  name: 'Nathalie Rivas',
  title: 'Hola poshito',
  subTitle: 'Desde tu mente :3'
}