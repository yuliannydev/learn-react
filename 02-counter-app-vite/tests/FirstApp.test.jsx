import React from "react";
import { render, screen } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe("testing FirstApp", () => {
  const title = 'Hola mundo';
  test("Debe hacer match con el Snapshot", () => {
    /*renderizado del componente*/
    const { container } = render(<FirstApp/>);
    /* Capture al componente */
    expect(container).toMatchSnapshot();
  });

  test('Debe contener el texto "Hola mundo"', () => { 

    render(<FirstApp title={title} />)
    expect(screen.getByText(title)).toBeTruthy()
   })

  test('Debe contener el texto "Hola mundo" en un h1', () => { 

    render(<FirstApp title={title} />)
    expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
   })

});
