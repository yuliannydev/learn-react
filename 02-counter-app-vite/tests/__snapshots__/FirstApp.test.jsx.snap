// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing FirstApp Debe hacer match con el Snapshot 1`] = `
<div>
  <h1>
     
    Hola poshito
     
  </h1>
  <p>
     
    Desde las defaultProps
     
  </p>
  <em>
     
    Nathalie Rivas
     
  </em>
</div>
`;
